client-spark/spark3/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -529,14 +538,25 @@ private Roaring64NavigableMap getExpectedTasksByExecutorId(
     } catch (Exception e) {
       throw new RssException(e);
     }
+    long expectedRecords = 0;
     while (mapStatusIter.hasNext()) {
       Tuple2<BlockManagerId, Seq<Tuple3<BlockId, Object, Object>>> tuple2 = mapStatusIter.next();
       if (!tuple2._1().topologyInfo().isDefined()) {
         throw new RssException("Can't get expected taskAttemptId");
       }
+      // Retrieve the validation info propagated from the shuffle writer
```

Comment:
Could you log more information? I think we should log more details to help data correctness issues.